common/src/warp_runner/manager/commands/constellation_commands.rs

```diff
@@ -373,31 +355,29 @@ fn go_back_to_previous_directory(
     get_items_from_current_directory(warp_storage)
 }
 
-async fn upload_files(
-    warp_storage: &mut warp_storage,
-    files_path: Vec<PathBuf>,
-    // todo: send FileTransferProgress::Step until done
-    tx: mpsc::UnboundedSender<FileTransferProgress<uplink_storage>>,
-) {
+async fn upload_files(warp_storage: &mut warp_storage, files_path: Vec<PathBuf>) {
+    let tx_upload_file = UPLOAD_FILE_LISTENER.tx.clone();
+
     let current_directory = match warp_storage.current_directory() {
         Ok(d) => d,
         Err(e) => {
-            let _ = tx.send(FileTransferProgress::Error(e));
+            let _ = tx_upload_file.send(UploadFileAction::Error(e));
             return;
         }
     };
 
     let max_size_ipfs = warp_storage.max_size();
     let current_size_ipfs = warp_storage.current_size();
-
-    for file_path in files_path {
+    for file_path in files_path.clone() {
         let mut filename = match file_path
             .file_name()
             .map(|file| file.to_string_lossy().to_string())
         {
             Some(file) => file,
             None => continue,
         };
+        let _ = tx_upload_file.send(UploadFileAction::Starting(filename.clone()));
```

Comment:
This should probably be moved until after the metadata function since this is indicating that a file is being uploaded but could silently fail. Moving it after `tokio::fs::metadata` or within the match condition might be the safer option in this case.